specs/waku/v2/waku-v2.md

```diff
@@ -147,7 +151,50 @@ packet from one version to the other.
+Waku v2 by default guarantees pseudonymity for all of the protocol layers since parties do not have to disclose their true identity and instead they utilize libp2p `PeerID` as their identifiers. While pseudonymity is an appealing security feature, it does not guarantee full anonymity since the actions taken under the same pseudonym i.e., `PeerID` can be linked together and potentially result in the re-identification of the true actor. 
+  
+### Anonymity / Unlinkability
+At a high level, Anonymity is the inability of an adversary in linking an actor to its data/performed action (the actor and action are context-dependent). To be precise about linkability, we use the term Personally Identifiable Information (PII) to refer to any piece of data that could potentially be used to uniquely identify a party. For example, the signature verification key, and the hash of one's static IP address are unique for each user and hence count as PII. Notice that users' actions can be traced through their PIIs (e.g., signatures) and hence result in their re-identification risk. As such, we seek anonymity by avoiding linkability between actions and the actors / actors' PII. Concerning anonymity, Waku v2 provides the following features:
+
+**Publisher-Message Unlinkability**: This feature signifies the unlinkability of a publisher to its published messages in the `WakuRelay` protocol. The [Publisher-Message Unlinkability](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-relay.md#security-analysis) is enforced through the `StrictNoSign` policy due to which the data fields of pubsub messages that count PII for the publisher must be left unspecified.
+
+**Subscriber-Topic Unlinkability**: This feature stands for the unlinkability of the subscriber to its subscribed topics in the `WakuRelay` protocol. The [Subscriber-Topic Unlinkability](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-relay.md#security-analysis) is achieved through the utilization of a single pubsub topic. As such, subscribers are not re-identifiable from their subscribed topic IDs as the entire network is linked to the same topic ID. This level of unlinkability / anonymity is known as [k-anonymity](https://www.privitar.com/blog/k-anonymity-an-introduction/) where k is proportional to the system size (number of subscribers). Note that there is no hard limit on the number of the pubsub topics, however, the use of one topic is recommended for the sake of anonymity. 
+
+### Spam protection
+This property indicates that no adversary can flood the system with spam messages (i.e., publishing a large number of messages in a short amount of time). Spam protection is partly provided in `WakuRelay` through [scoring mechanism](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#spam-protection-measures) of GossipSub v1.1. At a high level, peers utilize a scoring function to locally score the behavior of their connections and remove peers with a low score.   
```

Comment:
Spam, in this context, is flooding the system (either accidentally or intentionally) with _any_ kind of messages i.e. even if the message content is valid or useful. Also, it looks like GossipSub peer-scoring parameters are not frequency-based but there is a separate capping on `IHAVE/IWANT` messages. Might help to address some of that here to clarify what additional spam protection, if any, is Waku providing over that of GossipSub.

Reply:
(Btw, I've written about this topic here: https://our.status.im/can-we-can-spam/)

Reply:
>Spam, in this context, is flooding the system (either accidentally or intentionally) with any kind of messages i.e. even if the message content is valid or useful. 

Thanks! I revised the the definition of spam messages to avoid any ambiguity. 
> Might help to address some of that here to clarify what additional spam protection, if any, is Waku providing over that of GossipSub.

Waku does not yet provide any additional spam protection mechanism and relies on GossipSub, however, there is an ongoing development about the use of Rate Limiting Nullifiers in Waku Relay for this sake. Please see [RLN Relay](https://github.com/vacp2p/specs/blob/0bb2747b097da14f2544a2797988f8d5908a192a/specs/waku/v2/waku-rln-relay.md) for further details.  
Once the RLN-Relay gets finalized, will certainly provide a comparison between the spam protection method of GossipSub and Waku in RLN-Relay specs (and will reflect its summary in this spec as well). Thanks for your comment.

Reply:
Thanks for sharing your article. Added as the reference in the following issue https://github.com/vacp2p/specs/issues/286